drivers/timer/ifx_cat1_lp_timer.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2025 Cypress Semiconductor Corporation (an Infineon company) or
+ * an affiliate of Cypress Semiconductor Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Low Power timer driver for Infineon CAT1 MCU family.
+ */
+
+#define DT_DRV_COMPAT infineon_cat1_lp_timer
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/timer/system_timer.h>
+#include <zephyr/irq.h>
+#include <zephyr/spinlock.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/drivers/gpio.h>
+
+#include <cyhal_lptimer.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ifx_cat1_lp_timer, CONFIG_KERNEL_LOG_LEVEL);
+
+/* The application only needs one lptimer. Report an error if more than one is selected. */
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) > 1
+#error Only one LPTIMER instance should be enabled
+#endif
+
+#define LPTIMER_INTR_PRIORITY (3u)
+#define LPTIMER_FREQ          (32768u)
+
+/* We need to know the number of MCWDT instances.  Unfortunately, this information is not available
+ * in a header in the HAL code.  This was extracted from the cyhal_lptimer.c file in the HAL
+ */
+#if (defined(CY_IP_MXS40SRSS) || defined(CY_IP_MXS40SSRSS) || defined(CY_IP_MXS28SRSS) ||          \
+	defined(CY_IP_MXS22SRSS)) &&                                                               \
+	!((defined(CY_IP_MXS40SRSS) && (CY_IP_MXS40SRSS_VERSION >= 3)) ||                          \
+	((SRSS_NUM_MCWDT_B) > 0))
+#define NUM_LPTIMERS SRSS_NUM_MCWDT
+#else
+#error "Current HW block is not supported"
```

Comment:
```suggestion
#error "Selected device doesn't support low power timers at this time."
```
Let's make the error a little clearer to the user. They don't care about a "HW block", they care that they can or cannot use low power timers.

Reply:
Updated the error message